backend/core/views.py

```
@@ -963,103 +1018,68 @@ def quality_check_detail(self, request, pk):
+            return Response(
+                {"error": "Permission denied"}, status=status.HTTP_403_FORBIDDEN
+            )
+        risk_assessment = RiskAssessment.objects.get(id=pk)
+        risk_scenarios = risk_assessment.risk_scenarios.all()
+        queryset = AppliedControl.objects.filter(
+            risk_scenarios__in=risk_scenarios
+        ).distinct()
 
-            response = HttpResponse(content_type="text/csv")
+        # Use the same serializer to maintain consistency - to review
+        serializer = RiskAssessmentActionPlanSerializer(
```

Ruff (0.11.9)

`RiskAssessmentActionPlanSerializer` may be undefined, or defined from star imports

(F405)